src/combine.cpp

```diff
@@ -1416,6 +1416,54 @@ void do_symbol_check(chunk_t *prev, chunk_t *pc, chunk_t *next)
 
    if (language_is_set(LANG_CPP))
    {
+      chunk_t *nnext = chunk_get_next_ncnl(next);
+
+      if (  (pc->flags & (PCF_IN_STRUCT | PCF_IN_CLASS))
+         && chunk_is_token(pc, CT_ASSIGN)
+         && chunk_is_token(nnext, CT_SEMICOLON)
+         && (  chunk_is_token(next, CT_DEFAULT)
+            || chunk_is_token(next, CT_DELETE)
+            || (chunk_is_token(next, CT_NUMBER) && chunk_is_str(next, "0", 1))))
+      {
+         const size_t      level = pc->level;
+
+         static const auto end_cond_fcns = [](chunk_t *c, size_t lvl){
+            return(  c->level == lvl
+                  && (  c->type == CT_FUNC_CLASS_PROTO  // ctor/dtor
+                     || c->type == CT_FUNC_PROTO));     // normal function
+         };
+         static const auto end_cond_virtual = [](chunk_t *c, size_t lvl){
+            return(  c->level == lvl
+                  && c->type == CT_QUALIFIER
+                  && chunk_is_str(c, "virtual", 7));
+         };
+
+         auto const cond_fcn = chunk_is_token(next, CT_NUMBER)
+                               ? end_cond_virtual
+                               : end_cond_fcns;
+
+         bool    found_status = false;
+         chunk_t *pprev       = chunk_get_prev(pc);
+
+         for ( ; (  pprev != nullptr
+                 && pprev->level >= level
+                 && pprev->type != CT_SEMICOLON
+                 && pprev->type != CT_PRIVATE_COLON)
+               ; pprev = chunk_get_prev(pprev))
+         {
+            if (cond_fcn(pprev, level))
```

Comment:
I don't see any reason to do this with lambdas, a simple `if` `else` here would have been ok.